js/widgets/widgetWindows.js

```diff
@@ -343,8 +343,34 @@ window.widgetWindows.hideWindows = function(name) {
     });
 };
 
+window.widgetWindows.exitWindow = function(name) {
```

Comment:
This seems like it would be a nightmare to maintain. Cannot we figure out the name some other way? How does the widget itself generate the name?